### ISSUE TITLE
bug: NexMark: `drop materialized view` stuck

### ISSUE THREAD

**Issue by fuyufjh** [singularity-data/risingwave#4790]
**Describe the bug**

```sql
create source person
    (id INTEGER, name VARCHAR, "email_address" VARCHAR, "credit_card" VARCHAR, city VARCHAR, state VARCHAR, "date_time" TIMESTAMP)
with (
  connector = 'nexmark',
  nexmark.table.type = 'Person',
  nexmark.split.num = '9',
  nexmark.min.event.gap.in.ns = '0'
) row format json;

create source auction (id INTEGER, "item_name" VARCHAR, description VARCHAR, "initial_bid" INTEGER, reserve INTEGER, "date_time" TIMESTAMP, expires TIMESTAMP, seller INTEGER, category INTEGER)
with (
  connector = 'nexmark',
  nexmark.table.type = 'Auction',
  nexmark.split.num = '9',
  nexmark.min.event.gap.in.ns = '0'
) row format json;

create source bid (auction INTEGER, bidder INTEGER, price INTEGER, "date_time" TIMESTAMP)
with (
  connector = 'nexmark',
  nexmark.table.type = 'Bid',
  nexmark.split.num = '9',
  nexmark.min.event.gap.in.ns = '0'
) row format json;

create materialized view nexmark_q15 as
SELECT
     to_char(date_time, 'YYYY-MM-DD') as "day",
     count(*) AS total_bids,
     count(*) filter (where price < 10000) AS rank1_bids,
     count(*) filter (where price >= 10000 and price < 1000000) AS rank2_bids,
     count(*) filter (where price >= 1000000) AS rank3_bids,
     count(distinct bidder) AS total_bidders,
     count(distinct bidder) filter (where price < 10000) AS rank1_bidders,
     count(distinct bidder) filter (where price >= 10000 and price < 1000000) AS rank2_bidders,
     count(distinct bidder) filter (where price >= 1000000) AS rank3_bidders,
     count(distinct auction) AS total_auctions,
     count(distinct auction) filter (where price < 10000) AS rank1_auctions,
     count(distinct auction) filter (where price >= 10000 and price < 1000000) AS rank2_auctions,
     count(distinct auction) filter (where price >= 1000000) AS rank3_auctions
FROM bid
GROUP BY to_char(date_time, 'YYYY-MM-DD');
```

Then,

```sql
dev=> drop materialized view nexmark_q15; -- stuck forever
```

**To Reproduce**

as above

**Expected behavior**

Don't stuck.

**Additional context**

As shown in the dashboard, the actors are still alive...

<img width="1200" alt="image" src="https://user-images.githubusercontent.com/10192522/185896675-e8fab5c1-d461-45d5-938d-96e6e20ddb92.png">

**Comment by fuyufjh** [comment]
Seems not the problem of Q15. Still investigating...

**Comment by BugenZhao** [comment]
I'm also able to reproduce this locally with nexmark q7 with no CPU workload. The stack trace shows that all actors exited but the barrier-complete RPC stucks.

```
--- Actor Traces ---
--- RPC Traces ---
>> RPC 127.0.0.1:5688 (77)
[captured 38.465041ms ago]
/stream_service.StreamService/BarrierComplete:77 [109.302082165s]
  sync_epoch (epoch 2879141438816256) [!!! 109.293895999s]
    store_sync [!!! 109.293895999s]
```

cc @hzxa21 @wenym1 

```
dev=> create source bid (
    auction INTEGER, 
    bidder INTEGER, 
    price INTEGER, 
    date_time TIMESTAMP) 
with ( 
    connector = 'nexmark', 
    nexmark.table.type = 'Bid', 
    nexmark.split.num = '8',
    nexmark.min.event.gap.in.ns = '10000'
) row format JSON;
CREATE_SOURCE

dev=> CREATE MATERIALIZED VIEW nexmark_q7
AS
SELECT
    B.auction,
    B.price,
    B.bidder,
    B.date_time
from
    bid B
    JOIN (
        SELECT
            MAX(price) AS maxprice,
            window_end as date_time
        FROM
            TUMBLE(bid, date_time, INTERVAL '10' SECOND)
        GROUP BY
            window_end
    ) B1 ON B.price = B1.maxprice
WHERE
    B.date_time BETWEEN B1.date_time - INTERVAL '10' SECOND
    AND B1.date_time;
CREATE_MATERIALIZED_VIEW

dev=> drop materialized view nexmark_q7;
```

**Comment by BugenZhao** [comment]
Update: stuck at...

https://github.com/singularity-data/risingwave/blob/c275f7e0fc8de90f41de6361a7a7703f8f7cf05d/src/storage/hummock_sdk/src/filter_key_extractor.rs#L272

```
>> RPC 127.0.0.1:5688 (59)
[captured 19.485125ms ago]
/stream_service.StreamService/BarrierComplete:59 [14.602490207s]
  sync_epoch (epoch 2879211431264256) [!!! 13.654142791s]
    store_sync [!!! 13.654116583s]
      run_sync_upload_task [!!! 13.654052875s]
        shared_buffer_flush_compact [!!! 13.654052875s]
          compact_shared_buffer(2) [!!! 13.654052875s]
            filter_key_extractor_acquire [!!! 13.654052875s]
```

cc @Li0k

**Comment by hzxa21** [comment]
Fixed by #4798
